debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventQueue.java

```diff
@@ -209,28 +235,128 @@ public void close() {
+        private ObjectMapper mapper = new ObjectMapper();
+
+        public EventDedupCache(String dataDir) {
+            this.dataDir = dataDir;
+        }
+
+        public void warmUp(Map<Long, Long> totalEventsPerSegment) {
+            if (totalEventsPerSegment.size() == 0) {
+                LOGGER.info("No queue segments found. Nothing to recover into event dedup cache");
+                return;
+            }
+            cache = new HashSet<>();
+            long eventsToBeProcessed = 0;
+            // Find which segement to start reading from, the Map entries are in descending
+            // order of segment number
+            for (Map.Entry<Long, Long> entry : totalEventsPerSegment.entrySet()) {
+                currentQueueSegmentIndex = entry.getKey();
+                eventsToBeProcessed += entry.getValue();
+                if (eventsToBeProcessed >= maxCacheSize) {
+                    // We have enough events to fill the cache. Break out of loop
+                    break;
+                }
+            }
+            while (currentQueueSegmentIndex <= totalEventsPerSegment.size() - 1) {
+                currentQueueSegmentPath = getFilePathOfCurrentQueueSegment();
+                String line;
+                BufferedReader input;
+                try {
+                    input = new BufferedReader(new FileReader(currentQueueSegmentPath));
+                    while ((line = input.readLine()) != null) {
+                        if (line.equals(EOF_MARKER)) {
+                            break;
+                        }
+                        String eventId = getEventId(line);
+                        if (eventId == null) {
+                            continue;
+                        }
+                        this.addEventToCache(eventId);
+                    }
+                    input.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+                currentQueueSegmentIndex++;
+            }
+
+            LOGGER.info("Recovered {} events into event cache", this.getCacheSize());
+        }
+
+        private String getFilePathOfCurrentQueueSegment() {
+            String queueSegmentFileName = String.format("%s.%d.%s", QUEUE_SEGMENT_FILE_NAME, currentQueueSegmentIndex,
+                    QUEUE_SEGMENT_FILE_EXTENSION);
+            return Path.of(dataDir, QUEUE_FILE_DIR, queueSegmentFileName).toString();
+        }
+
+        public boolean isEventInCache(String event) {
+            return cache.contains(event.hashCode());
+        }
+
+        public void addEventToCache(String event) {
+            // Check if cache is full. If full, remove oldest event from cache and add new
+            // event
+            if (cache.size() >= maxCacheSize) {
+                String oldestEvent = mostRecentIdFirstList.removeLast();
+                cache.remove(oldestEvent.hashCode());
+            }
+            cache.add(event.hashCode());
```

Comment:
Why put hashCode()???? The `eventID` is unique -- why not add it directly to the cache?

Reply:
Yeah changed it